lib/MatterbridgeManager.php

```diff
@@ -642,6 +643,33 @@ public function getCurrentVersionFromBinary(): ?string {
 			return null;
 		}
 
+		// is www user the file owner?
+		$user = posix_getpwuid(posix_getuid());
+		$fileOwner = posix_getpwuid(fileowner($binaryPath));
+		$userIsOwner = $user['name'] === $fileOwner['name'];
+
+		// get file group and user groups
+		$fileGid = filegroup($binaryPath);
+		$myGids = posix_getgroups();
+
+		// check permissions
+		$perms = fileperms($binaryPath);
+		$uExec = (($perms & 0x0040) && !($perms & 0x0800));
+		$gExec = (($perms & 0x0008) && !($perms & 0x0400));
+		$aExec = (($perms & 0x0001) && !($perms & 0x0200));
+
+		// 3 ways to have the permission :
+		// * be the owner and have u+x perm
+		// * not be the owner, be in the file group and have g+x perm
+		// * have o+x perm
+		$execPerm = ($user['name'] === $fileOwner['name'] && $uExec)
+			|| ($user['name'] !== $fileOwner['name'] && in_array($fileGid, $myGids) && $gExec)
+			|| $aExec;
```

Comment:
```suggestion
		$execPerm = $aExec
			|| ($user['name'] === $fileOwner['name'] && $uExec)
			|| (in_array($fileGid, $myGids) && $gExec);
```

Sorted from easy comparison to hardest as complained by phpstorm

Reply:
Done. I kept the `$user['name'] !== $fileOwner['name']` associated with group permission because of previous explanation.